core/src/blocktree.rs

```diff
@@ -193,6 +193,46 @@ impl Blocktree {
         false
     }
 
+    // Helper that deletes columns that have `(Slot,Index)` as their Index
+    fn delete_indexed_column<C>(&self, column: &LedgerColumn<C>, delete_slot: u64) -> Result<()>
+    where
+        C: Column<Index = (u64, u64)>,
+    {
+        let iter = column.iter(Some((delete_slot, 0)))?;
+        let mut max_index = 0;
+
+        for ((slot, index), _) in iter {
+            if slot != delete_slot {
+                break;
+            }
+            max_index = index;
+        }
+        for i in 0..=max_index {
+            column.delete((delete_slot, i))?;
+        }
+        Ok(())
+    }
+
+    fn delete_slot(&self, slot: u64) -> Result<()> {
+        match self.meta(slot) {
```

Comment:
I don't think you need to predicate the delete_indexed_column() on whether or not meta is present for the slot...  delete with extreme prejudice, do not assume that meta is necessarily present, or truthy

Reply:
Yeah you're right. I think Im going to rework this PR to basically iter each column as far as possible and wipe it.